mix/qml/ProjectModel.qml

```diff
@@ -91,15 +93,44 @@ Item {
 
 	MessageDialog {
 		id: deployWarningDialog
+		property bool redeploy
 		title: qsTr("Project")
-		text: qsTr("This project has been already deployed to the network. Do you want to re-deploy it?")
-		standardButtons: StandardButton.Ok | StandardButton.Cancel
+		text:
+		{
+			if (Object.keys(projectModel.deploymentAddresses).length > 0)
+			{
+				redeploy = true
+				standardButtons = StandardButton.Ok | StandardButton.Reset | StandardButton.Abort;
+				return qsTr("This project has been already deployed to the network. Do you want to repackage the ressources only, or also reset the deployed contract to his initial state?")
```

Comment:
The developper may have made some modifications on ressources (html/js...) but not on the contract. So in that case we don't need to redeploy the contract.